core/src/main/python/wlsdeploy/tool/create/domain_creator.py

```diff
@@ -860,6 +861,17 @@ def __create_machines_clusters_and_servers(self, delete_now=True):
 
         cluster_nodes = dictionary_utils.get_dictionary_element(self._topology, CLUSTER)
         if len(cluster_nodes) > 0:
+            self.topology_helper.create_placeholder_clusters(self._topology)
+            # Frontend Host needs to be set before FrontEndHTTPPort as required by WLST
+            for cluster in cluster_nodes:
+                frontend_host = dictionary_utils.get_element(self._topology[CLUSTER][cluster], FRONTEND_HOST)
+                if frontend_host is not None:
+                    temp_loc = LocationContext()
+                    temp_loc.append_location(CLUSTER)
+                    temp_loc.add_name_token(self.aliases.get_name_token(temp_loc), cluster)
+                    attribute_path = self.aliases.get_wlst_attributes_path(temp_loc)
+                    self.wlst_helper.cd(attribute_path)
+                    self._set_attribute(temp_loc, FRONTEND_HOST, frontend_host, [])
```

Comment:
Have you considered overriding `creator._set_attributes()` in `domain_creator.py` to perform this check? That would save having to navigate the folders and create placeholders. There's a similar example where `domain_creator` overrides `_create_named_mbeans()`. There are a couple of calls needed to get the type name from the location, similar to `deployer.get_location_type()`.

Reply:
redesigned per request.